src/test/groovy/AxisStepTests.groovy

```diff
@@ -0,0 +1,68 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import org.junit.Before
+import org.junit.Test
+import java.lang.Exception
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertTrue
+
+class AxisStepTests extends ApmBasePipelineTest {
+  String scriptName = 'vars/axis.groovy'
+
+  @Override
+  @Before
+  void setUp() throws Exception {
+    super.setUp()
+  }
+
+  @Test
+  void test() throws Exception {
+    def script = loadScript(scriptName)
+    def v = script.call('foo', [1, 2])
+    printCallStack()
+    assertEquals(v.size(), 2)
+    // org.codehaus.groovy.runtime.GStringImpl<1> is not a java.lang.String<1>
```

Comment:
what's this?

Reply:
```suggestion
    // Remember to convert GStringImpl to String before compare it to a String
    // org.codehaus.groovy.runtime.GStringImpl<1> is not a java.lang.String<1>
```

Reply:
it is a Note about why the GStringImpl should be converted to String